spec/views/posts_index_spec.rb

```diff
@@ -0,0 +1,207 @@
+require 'rails_helper'
+
+# rubocop:disable Metrics/BlockLength
+
+RSpec.describe 'Posts index page', type: :system do
+  let!(:user) do
+    User.create!(
+      name: 'Bodia',
+      email: 'bodia@example.com',
+      password: 'password',
+      confirmed_at: Time.now,
+      posts_counter: 3,
+      bio: 'My name is Bodian, I love cooking and riding rollerblades'
+    )
+  end
+
+  let!(:posts) do
+    [
+      Post.create!(
+        author: user,
+        title: 'To jest tytył',
+        text: 'A to jest juz tekst',
+        likes_counter: 0,
+        comments_counter: 0
+      ),
+      Post.create!(
+        author: user,
+        title: 'This is a title',
+        text: 'This is text',
+        likes_counter: 0,
+        comments_counter: 0
+      ),
+      Post.create!(
+        author: user,
+        title: 'To jest tytył 2',
+        text: 'A to jest juz tekst 2',
+        likes_counter: 0,
+        comments_counter: 0
+      ),
+      Post.create!(
+        author: user,
+        title: 'This is a title 2',
+        text: 'This is text 2',
+        likes_counter: 0,
+        comments_counter: 0
+      )
+    ]
+  end
+
+  let!(:comments) do
+    [
+      Comment.create!(
+        author: user,
+        post: posts[1],
+        text: 'A to jest komentarz'
+      ),
+      Comment.create!(
+        author: user,
+        post: posts[1],
+        text: 'A to jest drugi komentarz'
+      ),
+      Comment.create!(
+        author: user,
+        post: posts[1],
+        text: 'A to jest trzeci komentarz'
+      ),
+      Comment.create!(
+        author: user,
+        post: posts[1],
+        text: 'A to jest czwarty komentarz'
+      ),
+      Comment.create!(
+        author: user,
+        post: posts[1],
+        text: 'A to jest piąty komentarz'
+      ),
+      Comment.create!(
+        author: user,
+        post: posts[1],
+        text: 'A to jest szósty komentarz'
+      )
+    ]
+  end
+
+  let!(:likes) do
+    [
+      Like.create!(post: posts[1], author: user),
+      Like.create!(post: posts[1], author: user),
+      Like.create!(post: posts[1], author: user),
+      Like.create!(post: posts[0], author: user)
+    ]
+  end
+
+  it 'checks if there is a profile image' do
+    visit '/users/sign_in'
+    fill_in 'user_email', with: user.email
+    fill_in 'user_password', with: user.password
+    click_button 'Log in'
```

Comment:
This logging in should be in some kind of `before(:each)` instead of copy pasting it into every test. This code is not following DRY rule